aws/resource_aws_db_instance.go

```diff
@@ -642,8 +642,11 @@ func resourceAwsDbInstanceCreate(d *schema.ResourceData, meta interface{}) error
 		}
 
 		if attr := d.Get("vpc_security_group_ids").(*schema.Set); attr.Len() > 0 {
-			modifyDbInstanceInput.VpcSecurityGroupIds = expandStringSet(attr)
-			requiresModifyDbInstance = true
+			var s []*string
+			for _, v := range attr.List() {
+				s = append(s, aws.String(v.(string)))
+			}
+			opts.VpcSecurityGroupIds = s
```

Comment:
Can you please use `expandStringSet()`? Thanks.

```suggestion
			opts.VpcSecurityGroupIds = expandStringSet(attr)
```